libs/model/src/middleware/authorization.ts

```diff
@@ -19,43 +40,107 @@ import { AddressAttributes, models } from '..';
  */
 
 /**
- * Finds one active community address that meets the arguments
+ * Binds actor to active community address that meets the arguments
  * @param actor command actor
  * @param payload command payload including aggregate id
  * @param roles roles filter
- * @returns authorized address or throws
  */
 const authorizeAddress = async (
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  { actor, payload }: CommandContext<any>,
+  { actor, payload }: CommandContext<CommandInput>,
   roles: Role[],
-): Promise<AddressAttributes> => {
-  if (!payload.id) throw new InvalidActor(actor, 'Must provide a community id');
-  if (!actor.address_id)
-    throw new InvalidActor(actor, 'Must provide an address');
+): Promise<z.infer<typeof Address>> => {
+  // By convention, secure requests must provide community_id/id + address arguments
+  const community_id =
+    ('community_id' in payload && payload.community_id) || payload.id;
+  if (!community_id)
+    throw new InvalidActor(actor, 'Must provide a community id');
+  if (!actor.address) throw new InvalidActor(actor, 'Must provide an address');
+
   // TODO: cache
   const addr = (
     await models.Address.findOne({
       where: {
         user_id: actor.user.id,
-        address: actor.address_id,
-        community_id: payload.id,
+        address: actor.address,
+        community_id,
         role: { [Op.in]: roles },
       },
       order: [['role', 'DESC']],
     })
   )?.get({ plain: true });
   if (!addr)
     throw new InvalidActor(actor, `User is not ${roles} in the community`);
+
+  // @ts-expect-error readonly property
```

Comment:
If `actor.addressId` is not readonly then it shouldn't be set to readonly. Using `@ts-expect-error` with `readonly property` will not narrow the type error catching to just that error. `@ts-expect-error` is the same as `@ts-ignore` except the latter will not throw a type error if the code doesn't also throw a type error. That is, if another type error occurs on `actor.addressId = addr.id;` e.g. `addressId` is removed, then `@ts-expect-error` will catch this error and we will not be made aware of the typing issue.

`@ts-expect-error` should be avoided like the plague.

Reply:
```
(actor as { addressId: number }).addressId = addr.id!;
```
I think this works to remove `readonly` and allows updating the `addressId` without using `@ts-expect-error`.